File: app/src/main/java/syed/com/wordrepo/activity/NewWordActivity.java
package syed.com.wordrepo.activity;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.text.TextUtils;
import android.widget.EditText;
import android.widget.Toast;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import syed.com.wordrepo.R;
import syed.com.wordrepo.WordRepoApplication;
import syed.com.wordrepo.entitiy.Word;
import syed.com.wordrepo.utility.Log;

public class NewWordActivity extends AppCompatActivity {
    //Todo implement eventbus need to cleanup
    public static final String EXTRA_REPLY = "NEW_WORD";

    @BindView(R.id.edit_word)
    public EditText mEditWordView;

    @BindView(R.id.edit_meaning)
    public EditText mEditMeaningView;

    @BindView(R.id.edit_description)
    public EditText mEditDescriptionView;

    @BindView(R.id.edit_note)
    public EditText mEditNoteView;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_new_word);
        ButterKnife.bind(this);
    }

    @OnClick(R.id.button_save)
    public void saveWord() {
        Log.info(this.getClass(),"Clicked on the save button");
        if (TextUtils.isEmpty(mEditWordView.getText())) {
            Toast.makeText(getApplicationContext(),"Word is empty. Please add a word", Toast.LENGTH_SHORT).show();
        } else {
            Word word = new Word(mEditWordView.getText().toString()
                    , mEditMeaningView.getText().toString()
                    , mEditDescriptionView.getText().toString()
                    , mEditNoteView.getText().toString());

            //Todo need to work with it. Not good to expose repository in the whole application context
            WordRepoApplication.getWordRepository().insertWord(word);
            finish();
//          EventBus.getDefault().post(word);
        }
    }
}

File: app/src/main/java/syed/com/wordrepo/activity/NewWordActivity.kt
package syed.com.wordrepo.activity

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.text.TextUtils
import android.view.View
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_new_word.*
import syed.com.wordrepo.R
import syed.com.wordrepo.WordRepoApplication
import syed.com.wordrepo.entitiy.Word
import syed.com.wordrepo.utility.Log

/**
 * Created by syed on 2/2/18.
 */

class NewWordActivity : AppCompatActivity() {

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_new_word)

        button_save.setOnClickListener { view -> saveWord(view) }
    }


    fun saveWord(view:View) {
        Log.info(NewWordActivity::class.java, "Clicked on the save button")
        if (TextUtils.isEmpty(edit_word.text)) {
            Toast.makeText(applicationContext, "Word is empty. Please add a word", Toast.LENGTH_SHORT).show()
        } else {
            val word = Word(null, edit_word.text.toString(), edit_meaning.text.toString(), edit_description.text.toString(), edit_note.text.toString())

            (applicationContext as WordRepoApplication).word = word
            setResult(RESULT_OK)
            finish()
            //          EventBus.getDefault().post(word);
        }
    }
}
